projects/igniteui-angular/src/lib/directives/mask/mask.directive.ts

```diff
@@ -23,7 +23,20 @@ export class IgxMaskDirective implements OnInit, AfterViewChecked, ControlValueA
      * ```
      */
     @Input('igxMask')
-    public mask: string;
+    public get mask(): string {
+        return this._mask || this.defaultMask;
+    }
+
+    public set mask(val: string) {
+        // B.P. 9th June 2021 #7490
+        const oldOptions = this.maskOptions;
+        if (val !== this._mask) {
+            this.setPlaceholder(val);
+            this._mask = val;
+        }
+        const cleanInputValue = this.maskParser.parseValueFromMask(this.inputValue, oldOptions);
```

Comment:
if you call this before change `_mask` you will not need to cache `this.maskOptions`. Also, why do not call this inside the if statement. Should we update the input value if mask did not change at all?